### PR TITLE
Abdulhakim/bug 2161532 request validation errors 400

### PR DESCRIPTION
**Pull request by hakimms**
[Bug 2161532](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2161532): Treat Only Response Validation Errors as 500

### CODE REVIEW

**Review by AlexanderSher — APPROVED** [review]
LGTM

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs [hunk 1]
```diff
@@ -63,6 +63,33 @@ public async Task TokenIssuanceStartObjectModelTest()
             Assert.True(TestHelper.DoesPayloadMatch(Payload.TokenIssuanceStart.ExpectedPayload, httpResponseMessage.Content.ReadAsStringAsync().Result));
         }
 
+        /// <summary>Test the request object to verify the correct HttpStatusCode is respond</summary>
+        [Test]
+        [TestCase(RequestStatusType.Successful, HttpStatusCode.OK)]
+        [TestCase(RequestStatusType.Failed, HttpStatusCode.BadRequest)]
+        [TestCase(RequestStatusType.ValidationError, HttpStatusCode.BadRequest)]
+        [TestCase(RequestStatusType.TokenInvalid, HttpStatusCode.Unauthorized)]
+        public async Task TokenIssuanceStartRequestValidationTest(RequestStatusType requestStatusType, HttpStatusCode httpStatusCode)
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs:72]: Do we already have a test for response validation to check for internal server error?

Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs:72]: We do have coverage to make sure response validations throw internal server error. In the TokenIssuanceStartActionTest test in MiscTests.cs. We test all actions and return InternalServerError for bad actions

**Comment by hakimms** [comment]
Example of Request Validation throwing 400

image

Example of Response Validation throwing 500 

image

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs (hunk 1 above)
Comment (HarmanDhunna) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs:72]: Interesting those weren't failing before.

Comment (HarmanDhunna) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs:72]: NIT: Next time either of us touch this file, let's change the name of this class.

Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs:72]: They weren't failing because it was always returning internal server error. It was the request validation logic that threw 500s when they should have been 400s. There was no coverage for this scenario before.

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 2]
```diff
@@ -42,7 +42,7 @@ public async override Task<AuthenticationEventResponse> Completed()
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), false);
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:45]: Our goal about using `ExceptionStatusCode` is so we don't have to pass true or false here. We should be able to look at the AuthenticationEventTriggerValidationException.ExceptionStatusCode in `Response.MarkAsFailed`

Comment (HarmanDhunna) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:45]: This might force you to update Signature in a couple of spots. We should make sure the only exceptions that we throw inside of the library are AuthenticationEventTriggerException so we always know what the statuscode is from the exception and not figure it out on the fly

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 3]
```diff
@@ -42,30 +42,44 @@ public async override Task<AuthenticationEventResponse> Completed()
...
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    // Response.MarkAsFailed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage));
+                    return await Failed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage)).ConfigureAwait(false);
                 }
                 else if (RequestStatus == RequestStatusType.TokenInvalid)
                 {
-                    Response.MarkAsUnauthorized();
+                    // Set response as unauthorized when token is invalid
+                    Response.StatusCode = System.Net.HttpStatusCode.Unauthorized;
```
Comment (robezirg) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:51]: if there is failed, why isn't there unauthorized?

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs [hunk 4]
```diff
@@ -93,13 +94,20 @@ public Task SetValueAsync(object result, CancellationToken cancellationToken)
...
-                    Response.Invalidate();
+                    try
+                    {
+                        Response.Validate();
+                        Response.Invalidate();
+                    }
+                    catch (ValidationException exception)
+                    {
+                        throw new AuthenticationEventTriggerResponseValidationException(exception.Message);
+                    }
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs:105]: Move this logic inside of `AuthenticationEventResponse.Validation()`

`AuthenticationEventResponse.Validation()` should know to throw the correct exception `AuthenticationEventTriggerResponseValidationException`

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 5]
```diff
@@ -42,30 +42,44 @@ public async override Task<AuthenticationEventResponse> Completed()
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    // Response.MarkAsFailed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage));
+                    return await Failed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage)).ConfigureAwait(false);
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:46]: In what scenario would status message be empty? I don't like the idea of us throwing a generic "failed".

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 6]
```diff
@@ -42,30 +42,44 @@ public async override Task<AuthenticationEventResponse> Completed()
...
 
             return Response;
         }
 
-        internal override Task<AuthenticationEventResponse> Failed(Exception exception, bool internalError)
+        /// <summary>
+        /// Sets the response statuscode, reason phrase and body when we want to return failed.
+        /// </summary>
+        /// <param name="exception"></param>
+        /// <returns></returns>
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:68]: Remove or add the comment for "return"

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 7]
```diff
@@ -42,30 +42,44 @@ public async override Task<AuthenticationEventResponse> Completed()
...
             {
                 Response = new TResponse();
             }
 
-            Response.MarkAsFailed(exception, internalError);
-            return Task.FromResult<AuthenticationEventResponse>((TResponse)Response);
+            Response.StatusCode = exception is AuthenticationEventTriggerValidationException ex ?
+                ex.ExceptionStatusCode
+                : System.Net.HttpStatusCode.InternalServerError;
+            Response.ReasonPhrase = string.Empty;
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:79]: NIT : https://www.freesoft.org/CIE/RFC/2068/43.htm

**Review by HarmanDhunna — CHANGES_REQUESTED** [review]

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 8]
```diff
@@ -42,30 +42,44 @@ public async override Task<AuthenticationEventResponse> Completed()
...
+            Response.Body = Helpers.GetFailedRequestPayload(exception);
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:80]: Should we call `GetFailedRequestPayload`, `GetFailedResponsePayload` instead?

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs [hunk 9]
```diff
@@ -63,6 +63,33 @@ public async Task TokenIssuanceStartObjectModelTest()
             Assert.True(TestHelper.DoesPayloadMatch(Payload.TokenIssuanceStart.ExpectedPayload, httpResponseMessage.Content.ReadAsStringAsync().Result));
         }
 
+        /// <summary>Test the request object to verify the correct HttpStatusCode is respond</summary>
+        [Test]
+        [TestCase(RequestStatusType.Successful, HttpStatusCode.OK)]
+        [TestCase(RequestStatusType.Failed, HttpStatusCode.BadRequest)]
+        [TestCase(RequestStatusType.ValidationError, HttpStatusCode.BadRequest)]
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs:70]: Did we have a scenario where if there are multiple things wrong with the response, we can aggregate them and throw all together instead of 1 by 1?

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs (hunk 5 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:45]: I'm not sure what scenarios would give a null or empty status message. But I will do some investigation into this.

Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:45]: This is a property that the end user can alter and possibly make it null or empty

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs (hunk 9 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/MiscTests.cs:70]: is this what you're referring to? [Bug 2433233](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2433233): Return all errors at once

**Comment by HarmanDhunna** [comment]
Please add this to contribution.md for beta6 and add release data

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 10]
```diff
@@ -42,30 +42,44 @@ public async override Task<AuthenticationEventResponse> Completed()
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    // Response.MarkAsFailed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage));
```
[the remark this answers is not in the document]
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs]: done

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md [hunk 11]
```diff
@@ -14,9 +14,12 @@
 
 ### Bugs Fixed
 
+- Made the request validation errors return 400 and the response validation errors return 500.
```
[the remark this answers is not in the document]
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md:17]: Done

**Review by HarmanDhunna — APPROVED** [review]

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 12]
```diff
@@ -42,30 +42,52 @@ public async override Task<AuthenticationEventResponse> Completed()
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    return await Failed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage)).ConfigureAwait(false);
```
Comment (AlexanderSher) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:45]: We usually have `Async` suffix for async methods. Why is it async btw?

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 13]
```diff
@@ -42,30 +42,52 @@ public async override Task<AuthenticationEventResponse> Completed()
...
+        public override Task<AuthenticationEventResponse> Failed(Exception exception)
         {
             if (Response == null)
             {
                 Response = new TResponse();
             }
 
-            Response.MarkAsFailed(exception, internalError);
-            return Task.FromResult<AuthenticationEventResponse>((TResponse)Response);
+            Response.StatusCode = exception is AuthenticationEventTriggerValidationException ex ?
```
Comment (AlexanderSher) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:72]: Please group this into one if/else check

Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs]: done

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs (hunk 12 above)
Comment (HarmanDhunna) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:45]: I believe it's just how it was originally there. I don't see any reason of keeping it async. Our SDK doesn't make any external calls. They user of SDK could be doing that but we don't need to have this to be async.

@hakimms It's something we should definitely tackle when we move this to be an isolated process.

Comment (HarmanDhunna) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:45]: Actually... i take it back, we have external call for TokenValidation, but happens before this. Regardless, this particular method doesn't need to be async

Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs]: Removed async for Failed and Completed.

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs [hunk 14]
```diff
@@ -3,6 +3,7 @@
 
 using System;
 using System.Buffers;
+using System.ComponentModel.DataAnnotations;
```
Comment (robezirg) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs:6]: why is this needed?

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs [hunk 15]
```diff
@@ -74,7 +102,7 @@ public async Task TokenIssuanceStartObjectModelNullResponseTest()
                 {
                     request.Response = null;
 
-                    eventsResponseHandler.SetValueAsync(request.Completed().Result, CancellationToken.None);
+                    eventsResponseHandler.SetValueAsync(request.Completed(), CancellationToken.None);
```
Comment (robezirg) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs:105]: why us this async?

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 16]
```diff
@@ -36,36 +36,64 @@ public abstract class AuthenticationEventRequest<TResponse, TData> : Authenticat
...
                 else if (RequestStatus == RequestStatusType.TokenInvalid)
                 {
-                    Response.MarkAsUnauthorized();
+                    Unauthorized();
                 }
             }
             catch (Exception ex)
             {
-                return await Failed(ex, true).ConfigureAwait(false);
+                return Failed(new AuthenticationEventTriggerRequestValidationException(ex.Message));
```
Comment (robezirg) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:54]: I don't think this should be a validation exception

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs (hunk 15 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs:105]: This object implements the interface IValueBinder which requires this implementation.

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs (hunk 16 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs]: Updated to just Exception

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs (hunk 14 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventResponseHandler.cs]: removed

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs (hunk 15 above)
Comment (robezirg) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs:105]: what does SetValueAsync return? If it is a task, are we intentionally not awaiting it?

Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs]: Updated to await

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs [hunk 17]
```diff
@@ -23,13 +23,13 @@ public void ConcurrencyTest()
...
+                    HttpResponseMessage httpResponseMessage = await EventResponseBaseTest(async eventsResponseHandler =>
                     {
-                        eventsResponseHandler.SetValueAsync(Payload.TokenIssuanceStart.ActionResponse, CancellationToken.None);
+                        await eventsResponseHandler.SetValueAsync(Payload.TokenIssuanceStart.ActionResponse, CancellationToken.None);
                     });
 
-                    Assert.AreEqual(System.Net.HttpStatusCode.OK, httpResponseMessage.StatusCode);
-                    Assert.True(TestHelper.DoesPayloadMatch(Payload.TokenIssuanceStart.ExpectedPayload, httpResponseMessage.Content.ReadAsStringAsync().Result));
+                    Assert.AreEqual(HttpStatusCode.OK, httpResponseMessage.StatusCode);
+                    Assert.True(DoesPayloadMatch(Payload.TokenIssuanceStart.ExpectedPayload, httpResponseMessage.Content.ReadAsStringAsync().Result));
```
Comment (AlexanderSher) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs:32]: Please await on `ReadAsStringAsync()` before using it in assert

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventResponse.cs [hunk 18]
```diff
@@ -96,25 +96,23 @@ internal void SetJsonValue<T>(Dictionary<string[], T> values)
...
-
-        internal void MarkAsUnauthorized()
-        {
-            StatusCode = System.Net.HttpStatusCode.Unauthorized;
-            ReasonPhrase = String.Empty;
-            Body = String.Empty;
+            try
+            {
+                Helpers.ValidateGraph(this);
+                Invalidate();
```
Comment (AlexanderSher) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventResponse.cs:110]: Why do we call `Invalidate` inside `Validate` method?

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs [hunk 19]
```diff
@@ -36,36 +36,64 @@ public abstract class AuthenticationEventRequest<TResponse, TData> : Authenticat
...
+
+            Response.Body = Helpers.GetFailedResponsePayload(exception);
+
+            return Response;
+        }
+
+        /// <summary>
+        /// Sets the response statuscode to unauthorized and the reasonphrase and body to empty.
+        /// </summary>
+        private void Unauthorized()
```
Comment (AlexanderSher) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs:91]: Let's call it `MakeResponseUnauthorized`, so it is clear from the name what method is doing

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Helpers.cs [hunk 20]
```diff
@@ -68,7 +68,12 @@ internal static HttpResponseMessage HttpErrorResponse(Exception ex)
             return response;
         }
 
-        internal static string GetFailedRequestPayload(Exception ex)
+        /// <summary>
+        /// Joins the exception messages into a json payload.
+        /// </summary>
+        /// <param name="ex">The exception thrown</param>
```
Comment (AlexanderSher) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Helpers.cs:74]: It is not obvious that exception is allowed to be null and what method would do if it is null. Please add this information to the comment.

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs (hunk 19 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs]: Updated

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Helpers.cs (hunk 20 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Helpers.cs]: Updated

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs (hunk 17 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs]: Done

Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/RequestResponseModelTests.cs]: I have updated the tests that use ReadAsStringAsync to be awaited. And I will create an ADO work item to update them in `AuthenticationEventResponseHandler.cs` and `AuthenticationEventResponse.cs`

**Review by AlexanderSher — APPROVED** [review]

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Common/Exceptions/AuthenticationEventTriggerRequestValidationException.cs [hunk 21]
```diff
@@ -12,7 +12,10 @@ namespace Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents
     internal class AuthenticationEventTriggerRequestValidationException : AuthenticationEventTriggerValidationException
     {
         /// <inheritdoc/>
-        public override HttpStatusCode ExceptionStatusCode => HttpStatusCode.InternalServerError;
+        public override HttpStatusCode ExceptionStatusCode => HttpStatusCode.BadRequest;
+
+        /// <inheritdoc/>
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Common/Exceptions/AuthenticationEventTriggerRequestValidationException.cs:17]: For later, we should see if there is a common library that we can use to pull these strings given the exception status code.

Comment (HarmanDhunna) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Common/Exceptions/AuthenticationEventTriggerRequestValidationException.cs:17]: NOT FOR THIS PR. Just a note for us.

**Review by HarmanDhunna — APPROVED** [review]
